responsibleai/responsibleai/_interfaces.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+from enum import Enum
+from typing import List
+
+
+class Dataset(object):
+    predictedY: List
+    features: List[List[int]]
+    featureNames: List[str]
+    probabilityY: List
+    trueY: List
+    classNames: List[str]
+
+
+class BoundedCoordinates(object):
+    type: str
+    names: List
+    scores: List
+    scores_range: List
+    upper_bounds: List
+    lower_bounds: List
+
+
+class EBMGlobalExplanation (object):
+    feature_list: List[BoundedCoordinates]
+
+
+class FeatureImportance(object):
+    scores: List
+    intercept: float
+    featureNames: List[str]
+
+
+class PrecomputedExplanations(object):
+    localFeatureImportance: FeatureImportance
+    globalFeatureImportance: FeatureImportance
+    ebmGlobalExplanation: EBMGlobalExplanation
+    customVis: str
+
+
+class ModelClass(str, Enum):
+    Tree = "Tree"
+    EBM = "EBM"
+    blackbox = "blackbox"
+
+
+class ModelMethod(str, Enum):
+    classifier = "classifier"
+    regressor = "regressor"
+
+
+class ModelExplanationData(object):
+    modelClass: ModelClass
+    method: ModelMethod
+    predictedY: List
+    probabilityY: List[List[float]]
+    explanationMethod: str
+    precomputedExplanations: PrecomputedExplanations
+
+
+class ErrorAnalysisData(object):
+    maxDepth: int
+    numLeaves: int
+
+
+class CausalMetric(object):
+    ci_lower: float
+    ci_upper: float
+    feature: str
+    p_value: float
+    point: float
+    stderr: float
+    zstat: float
+
+
+class CausalData(object):
+    globalCausalEffects: List[CausalMetric]
```

Comment:
All python objects should use snake case `global_causal_effects`

Reply:
I didn't comment this when this code was in raiwidgets, but now that it's in responsibleai it should be proper python style

Reply:
There will be hundreds of file changed on ui side, will send a separate pr for it today.